### DISCUSSION TITLE
Interest in a healthcheck how-to for containers that are connected to gluetun?

### DISCUSSION THREAD

**Discussion by ioqy** [qdm12/gluetun#1728] [Ideas]
There seem to be some people (including myself) that have run into the problem that the containers connected to gluetun are loosing their connection after gluetun reconnects:
https://github.com/qdm12/gluetun/issues/1277
https://github.com/qdm12/gluetun/issues/1360
https://github.com/qdm12/gluetun/issues/1407
https://github.com/qdm12/gluetun/issues/1441

I'm sure I have found a relative simple solution for this problem that only requires adding a healthcheck command to the connected container and adding the [willfarrell/autoheal](https://github.com/willfarrell/docker-autoheal) container.

And before I am putting any time in writing a how-to for the wiki, I wanted to know from @qdm12 if there is an interest for this?

**Comment by qdm12** [comment]
Is this problem happening after a VPN reconnection within the same gluetun container, or after a full Gluetun container restart?

**Reply by ioqy** [reply]
The problem is happening in both instances. But I just discovered that a restart of the gluetun container behaves differently from a reconnection of the VPN within the gluetun container. On a gluetun container restart the container which is connected to gluetun looses all connections. But on a VPN reconnect only the open ports of the application within the connected container stop working.

**Reply by qdm12** [reply]
The first instance (container restart) subscribe to #641

The second one is interesting though, so listening ports then no longer work? Or is it ongoing connections? Ongoing outbound connections will stop working this is expected, for example an http client would need to close its re-usable tcp connections.

**Reply by ioqy** [reply]
For the first instance (container restart) I have a working fix with the added health check and the autoheal container. But as I today found out, this doesn't cover the second instance.

On the second instance the listening port for incoming connections of Deluge is no longer working.
